### PR TITLE
add writer_batch_size to GeneratorBasedBuilder

### PR DESCRIPTION
**Pull request by lhoestq**
You can now specify `writer_batch_size` in the builder arguments or directly in `load_dataset`

### CODE REVIEW

**Comment by patrickvonplaten** [comment]
Awesome that's great!

File: src/nlp/builder.py [hunk 1]
```diff
@@ -556,6 +556,10 @@ class GeneratorBasedBuilder(DatasetBuilder):
     (`_split_generators`). See the method docstrings for details.
     """
 
+    def __init__(self, *args, **kwargs):
+        super(GeneratorBasedBuilder, self).__init__(*args, **kwargs)
+        self._writer_batch_size = kwargs.get("writer_batch_size")
```
Comment (patrickvonplaten) [inline src/nlp/builder.py:561]: maybe it's nice to use `DEFAULT_MAX_BATCH_SIZE` here as a default for the `.get()` function,  but both ways are fine with me!

Comment (lhoestq) [reply src/nlp/builder.py:561]: I think this parameter is more linked to the implementation of the `ArrowWriter`, that's why I think it's better if the default value is set in the writer itself

Comment (patrickvonplaten) [reply src/nlp/builder.py:561]: Fair!

**Pull request merged**